pandas/tests/resample/test_period_index.py

```diff
@@ -790,7 +790,7 @@ def test_resample_with_only_nat(self):
         frame = DataFrame([2, 3, 5], index=pi)
         expected_index = PeriodIndex(data=[], freq=pi.freq)
         expected = DataFrame(index=expected_index)
-        result = frame.resample("1s").mean()
+        result = frame.resample("1s", group_keys=False).mean()
```

Comment:
I would expect for a direct reduction method like `.mean()` (not a generic `apply()`) that such a `group_keys=False` is not needed to avoid a warning/behaviour change? (and actually would expect it to not have any effect)

Reply:
Thanks for finding this. What happens is `Resampler._groupby_and_aggregate` attempts to call aggregate, fails because there are no groups, and calls apply as a last resort. It seems best to me to make aggregate handle an empty grouper, but not sure how feasible that is (empty objects are typically rabbit holes...). I'll see if I can resolve it this way, but if there are other suggestions of how to fix, they are most welcome.